stripe_test.go

```diff
@@ -149,142 +149,188 @@ func TestShouldRetry(t *testing.T) {
 	).(*BackendImplementation)
 
 	// Exceeded maximum number of retries
-	assert.False(t, c.shouldRetry(
-		nil,
-		&http.Request{},
-		&http.Response{},
-		int(MaxNetworkRetries),
-	))
+	t.Run("DontRetryOnExceededRetries", func(t *testing.T) {
+		shouldRetry, _ := c.shouldRetry(
```

Comment:
I'm not checking against the returned message because:

* The boolean is by far the more important value.
* Messages aren't super likely to regress in an important way.
* Not checking the messages means that it's easier for us to change one because we don't have to update the test case as well.